### PR TITLE
Better craftable FBP Drones.

### PR DESCRIPTION
**Pull request by r4d6**
## About The Pull Request
This add better variations of the SI Sword Drone and SI Mantis drone that need to be manually crafted by a roboticist.
Also add an admin-only rocket hammer drone, because HOUND sprited them too.
Thanks to Nayu for spriting the fabrication steps for the Sword and Mantis Drones.

Misc. Change
- The roomba circuit board is now the custom circuit board, made using any circuit board, some plastic, gold & silver.
- The CRO can make robot stuff too.
- The robot recipes are now tied to the 'Robotic Expert' perk instead of the job slot.
- The Sword Drone, Mantis Drone, SI Roomba, Armored Roomba and 7K-Ten can now be repaired by welding them with help intent.
- Sword and Mantis Drone now drop the cell use in their crafting on death.
- Roombas are no longer dense and will no longer be able to block all bullets despite being shorter than a table.
- You can now sit on custom roombas.
- Roombas will no longer move when being dragged
- Making drones and roomba now show what you do to others.
- Added Fabrication Kits which are available in the Robotech Deluxe, they contain all the parts needed to make a Roomba/Sword/Mantis Drone, with the Roomba costing 250 and the sword/mantis costing 750. Prices may be adjusted later on.
- The armor plating and Heavy armor plating now give more HP to the roomba in addition to more armor.

Fabrication recipe : 
First, you'll need to make the Main Frame of the drone you want to build, this need steel, plasteel, and a bit of tools.
Second, you'll need the limbs, made with steel, plasteel and actuators, and the head, with steel, plasteel & a robot camera.
Then wrench & tighten the bolts with a wrench and a screwdriver.
Add the hydraulic system before securing it with a wrench.
Add either one or two armblades depending on if you're making a Sword Drone or a Mantis Drone.
Add any kind of Large Power Cell and secure it with a screwdriver.
Add the custom control board and secure that one too.
Wire up the drone using cables.
Program the drone's circuit board using a multitool.
Close the panel using a screwdriver.
Add and weld a 'Mark II' environmental protection suit.
Paint the drone red with Red Paint.
Swipe an ID with Robotic Access to activate the drone and actually spawn one in-game.

### CODE REVIEW

**Comment by lolman360** [comment]
> Roombas will no longer move when being dragged

whyyy

**Comment by r4d6** [comment]
> > Roombas will no longer move when being dragged
> 
> whyyy

That mean the roombas won't wander out of your grasp while you're draging them, AKA won't move right or left while you are moving.

File: code/modules/mob/living/carbon/superior_animal/drone/life.dm [hunk 1]
```diff
@@ -0,0 +1,7 @@
+/mob/living/carbon/superior_animal/handmade/Life()
+	. = ..()
+
+	if((following) && !(findTarget())) // Are we following someone and not attacking something?
+		while(get_dist(src, following) >= 3) // Keep walking until we get there.
+			step_to(src, following) // Take a step toward the friend
```
Comment (MSieghardt) [inline code/modules/mob/living/carbon/superior_animal/drone/life.dm:6]: Remove the while(get_dist(src, following) >= 3) and step_to.

Replace with walk_to(src, following, 4, 5)

The 4 is the distance you have to people (1 = next to them)
The 5 is the speed (experiment with it)

File: code/modules/mob/living/carbon/superior_animal/drone/handmade_drone.dm [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	attacktext = "slashed"
+	meat_amount = 0
+	mob_size = MOB_MEDIUM
+	randpixel = 0
+
+	var/obj/item/weapon/cell/large/cell = new /obj/item/weapon/cell/large/moebius // Hold the drone's power cell, default to a cheap one.
+	var/mob/following = null // Who are we following?
+
+/mob/living/carbon/superior_animal/handmade/death()
+	..()
```
Comment (MSieghardt) [inline code/modules/mob/living/carbon/superior_animal/drone/handmade_drone.dm:22]: Change the death()
To this
```
/mob/living/carbon/superior_animal/handmade/death(var/gibbed, var/message = "blows apart!")
	if (stat != DEAD)
		new /obj/effect/decal/cleanable/blood/gibs/robot(src.loc)
	
		if(cell) // Only if it does have a cell
			cell.forceMove(src.loc) // Drop the power cell
			cell = null // No more cell in the drone
		
		.=..()
		if(src)
			qdel(src)
```

**Review by MSieghardt — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/superior_animal/drone/life.dm (hunk 1 above)
Comment (r4d6) [reply code/modules/mob/living/carbon/superior_animal/drone/life.dm:6]: Thanks a lot!

File: code/modules/mob/living/carbon/superior_animal/drone/handmade_drone.dm (hunk 2 above)
Comment (r4d6) [reply code/modules/mob/living/carbon/superior_animal/drone/handmade_drone.dm:22]: Thanks a lot!

**Pull request merged**